doc/man3/flux_future_promise.adoc

```diff
@@ -0,0 +1,163 @@
+flux_future_promise(3)
+======================
+:doctype: manpage
+
+
+NAME
+----
+flux_future_promise, flux_future_set_result, flux_future_set_error, flux_future_aux_get, flux_future_aux_set, flux_future_set_flux, flux_future_get_flux - support methods for future wrappers
+
+
+SYNOPSIS
+--------
+ #include <flux/core.h>
+
+ typedef void (*flux_promise_f)(flux_future_t *f,
+                                flux_reactor_t *r, void *arg);
+
+ void flux_future_promise (flux_future_t *f,
+                           flux_promise_f cb, void *arg);
+
+ void flux_future_set_result (flux_future_t *f,
+                              void *result, flux_free_f free_fn);
+
+ void flux_future_set_error (flux_future_t *f, int errnum);
+
+ void *flux_future_aux_get (flux_future_t *f, const char *name);
+
+ int flux_future_aux_set (flux_future_t *f, const char *name,
+                          void *aux, flux_free_f destroy);
+
+ void flux_future_set_flux (flux_future_t *f, flux_t *h);
+
+ flux_t *flux_future_get_flux (flux_future_t *f);
+
+
+DESCRIPTION
+-----------
+
+A Flux future represents some activity that may be completed with reactor
+watchers and/or message handlers.  It is intended to be wrapped
+by other classes that perform specific activities, providing a common idiom
+for synchronizing on the result.  Typically such a class provides
+class-specific wrappers for `flux_future_create()` and `flux_future_get()`,
+and implements a class-specific promise callback.
+
+`flux_future_promise()` registers a promise callback.  The promise sets
+up class-specific watchers on the reactor to fulfill the promise as
+asynchronous events occur.  The watchers must eventually call
```

Comment:
It took me a bit to come to grips with the use of the term "fulfilling the promise" here. I would have preferred "fulfill the future" since the future is the object representing the unknown/deferred result.

Though there are examples in other languages, e.g. C++, of use of "fulfill the promise", this seems to be when a promise is a full-fledged object, and acts as the provider side handle of a future-promise object (so resolution of the future is a side effect of promise::fulfill.)

I was also literally going off this quote from [Wikipedia](https://en.wikipedia.org/wiki/Futures_and_promises), though I'm not sure how much weight to give it

> Setting the value of a future is also called *resolving*, *fulfilling*, or *binding* it.

Though reading through the manpage, I did kind of get where you were going "fulfill the promise" as meaning that the *promise callback* has completed.

I think in the end as long as we're consistent it will be fine, however I just wanted to note my initial trouble with terming `flux_future_set_result` as "fulfilling the promise".